ts/src/base/Exchange.ts

```diff
@@ -2315,33 +2327,42 @@ export default class Exchange {
          * @returns {object | undefined}
          */
         const value = this.safeValueN (dictionaryOrList, keys, defaultValue);
-        if (value === undefined) {
-            return defaultValue;
-        }
-        if ((typeof value === 'object') && !Array.isArray (value)) {
+        if ((value !== undefined) && (typeof value === 'object') && !Array.isArray (value)) {
```

Comment:
this line is not performance related, just made it inliner (there were unnecessary checkings),